services/web/server/src/simcore_service_webserver/payments/_db.py

```diff
@@ -121,19 +128,50 @@ async def get_pending_payment_transactions_ids(app: web.Application) -> list[Pay
 
 
 async def complete_payment_transaction(
-    app: web.Application, *, payment_id: PaymentID, success: bool, error_msg: str | None
+    app: web.Application,
+    *,
+    payment_id: PaymentID,
+    completion_state: PaymentTransactionState,
+    state_message: str | None,
 ) -> PaymentsTransactionsDB:
+    """
+
+    Raises:
+        PaymentNotFoundError
+
+    """
+    if completion_state == PaymentTransactionState.PENDING:
+        raise ValueError(f"{completion_state} is not a completion state")
+
     optional = {}
-    if error_msg:
-        optional["errors"] = error_msg
+    if state_message:
+        optional["state_message"] = state_message
 
     async with get_database_engine(app).acquire() as conn:
-        result = await conn.execute(
-            payments_transactions.update()
-            .values(completed_at=func.now(), success=success, **optional)
-            .where(payments_transactions.c.payment_id == payment_id)
-            .returning(literal_column("*"))
-        )
-        row = await result.first()
-        assert row  #  nosec
-        return PaymentsTransactionsDB.parse_obj(dict(row.items()))
+        async with conn.begin():
+            # TODO: test concurrency here
```

Comment:
TODO??
concurrency works if you acquire for each coroutine a connection. if you don't then it will fail. unless you do a batched call where you get multiple rows right away